redeffect.mlg

```diff
@@ -21,46 +20,45 @@ DECLARE PLUGIN "redeffect_plugin"
 (** *******************************************************************)
 
 {
-let check_ID_type typ =
-  let env = Global.env () in
-  let sigma = Evd.from_env env in
-  let decls,_ = splay_prod env Evd.empty typ in
+let check_ID_type env sigma typ =
+  let decls,_ = splay_prod env sigma typ in
   match List.rev decls with
-  | (_,t)::(_,u)::_
-     when
-       let t = EConstr.to_constr Evd.empty t in
-       let u = EConstr.to_constr Evd.empty u in
-       is_Type (Reduction.whd_all env t) &&
-       isRelN 1 (Reduction.whd_all env u) -> ()
+  | (x,t)::(_,u)::_
+    when
+      EConstr.isType sigma (whd_all env sigma t) &&
+      let xt = Context.Rel.Declaration.LocalAssum (x,t) in
+      EConstr.isRelN sigma 1 (whd_all (EConstr.push_rel xt env) sigma u) -> ()
   | _ ->
-    CErrors.user_err Pp.(strbrk "Printing function should be of type \"forall A:Type, A -> T\" for some T and it will print its second argument." ++ Printer.pr_econstr_env env sigma typ)
+    CErrors.user_err Pp.(
+        strbrk ("Printing function should be of type \"forall A:Type, A -> T\" "^
+                "for some T and it will print its second argument, instead of type")
```

Comment:
```suggestion
        strbrk ("Function printing its second argument should be of "^
                "type \"forall A:Type, A -> T\" for some T, instead of type")
```